src/examples/signal_generator.cpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * @file signal_generator.cpp
+ * @author Luciana Moreno Borges (youlumoreno.borges@gmail.com)
+ * @brief Configure the components, connect them and start the state machine
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <memory>
+#include <QApplication>
+#include <QTimer>
+#include <hyro/core/StateMachine.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+#include <hyro/widgets/WidgetCollection.h>
+#include <hyro/utils/StateMachineSpinner.h>
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/DigitalConverterComponent.h>
+
+using namespace hyro;
+/**
+ * @brief Main loop
+ * 
+ * @param argc 
+ * @param argv 
+ * @return int 
+ */
+
+void dynamic_loop(){
+  
+  /* Initialize dynamic properties with a default value */
+  float amplitude = 0.0f;
```

Comment:
Initialization of properties is either through the ComponentConfiguration, or with a default value present in the `init` method, like ` m_amplitude = config.parameters.getParameter<float>("amplitude", 3.0);`

You can remove them from here.